packages/components/organisms/f-form/src/components/_tests/FormField.test.js

```diff
@@ -0,0 +1,459 @@
+                        isEmpty: () => isEmpty,
+                        validations: () => ({ email: fieldValidation })
+                    },
+                    provide: () => ({
+                        $v
+                    })
+                });
+
+                // Assert
+                expect(wrapper.vm.hasInvalidError).toEqual(expected);
+            });
+        });
+
+        describe('errorType ::', () => {
+            it.each([
+                [false, true, null],
+                [false, false, null],
+                [true, false, 'invalid'],
+                [true, true, 'required']
+            ])('when `hasError` is %s and `hasRequiredError` is %s, should return %s', (hasError, hasRequiredError, expected) => {
+                // Arrange && Act
+                const wrapper = shallowMount(FormField, {
+                    propsData,
+                    computed: {
+                        hasError: () => hasError,
+                        hasRequiredError: () => hasRequiredError
+                    },
+                    provide: () => ({
+                        $v
+                    })
+                });
+
+                // Assert
+                expect(wrapper.vm.errorType).toEqual(expected);
+            });
+        });
+
+        describe('ariaLabel ::', () => {
+            it('should return the mobile number with spaces after every character', () => {
+                const expectedMobileNumber = '+ 4 4 7 1 1 1 1 1 1 1 1 1';
+
+                // Act
+                const wrapper = shallowMount(FormField, {
+                    propsData: {
+                        ...propsData,
+                        fieldData: {
+                            ...propsData.fieldData,
+                            value: '+447111111111',
+                            name: 'mobileNumber'
+                        }
+                    },
+                    provide: () => ({
+                        $v
+                    })
+                });
+
+                // Assert
+                expect(wrapper.vm.ariaLabel).toEqual(expectedMobileNumber);
+            });
+        });
+    });
+
+    describe('methods ::', () => {
+        afterEach(() => {
+            jest.clearAllMocks();
+        });
+
+        describe('updateField ::', () => {
+            it('should emit `updated` event with fieldName and field value', () => {
+                // Arrange
+                const wrapper = shallowMount(FormField, { propsData, provide: { $v } });
+                const payload = { fieldName: 'firstName', value: 'Joe' };
+
+                // Act
+                wrapper.vm.updateField(payload);
+
+                // Assert
+                expect(wrapper.emitted('updated').length).toBe(1);
+                expect(wrapper.emitted('updated')[0][0]).toEqual(payload);
+            });
+        });
+
+        describe('formFieldBlur ::', () => {
+            it('should call `touch` for validation', () => {
```

Comment:
Again, it's probably worth testing the result (form field is now dirty) rather than the implementation (`touch` method has been called).

Reply:
See other comment on other thread